R/util_dfc.R

```diff
@@ -118,13 +118,178 @@ dfc.compare_df_values <- function(df1, df2, id_col, verbose = FALSE) {
   # need to attach the original ids too.
   dfdiff_sum_rows[, id_col] <- df1[, id_col]
   dfdiff_sum_rows <- dfdiff_sum_rows[, c(id_col, "pct_unmatched", "num_unmatched")]
-
+  
+  # get a side-by-side comparison of any unmatched values
+  unmatched_summary <- dfdiff_sum_cols[dfdiff_sum_cols$num_unmatched > 0, ]
+  unmatched_cols <- unmatched_summary$variable_name
+  
+  if(length(unmatched_cols) > 0){
+    side_by_side_df <- data.frame(matrix(nrow = nrow(dfdiff_sum_rows), ncol = 0))
+    # we know they're all sorted the same way, so we can build the data.frame just by concatenating columns
+    side_by_side_df[[id_col]] <- dfdiff_sum_rows[[id_col]]
+    for(col in unmatched_cols){
+      side_by_side_df[[col %+% "_matches"]] <- dfdiff[[col]]
+      side_by_side_df[[col %+% "_df1"]] <- df1[[col]]
+      side_by_side_df[[col %+% "_df2"]] <- df2[[col]]
+    }
+  } else{
+    side_by_side_df <- NA
+  }
+  
   if(verbose) {util.html_table(head(dfdiff_sum_rows))}
 
   # Return diff DF and full summary dfs for user.
   return(list(diff_df = dfdiff,
               row_summary = dfdiff_sum_rows,
-              col_summary = dfdiff_sum_cols))
+              col_summary = dfdiff_sum_cols,
+              side_by_side_df = side_by_side_df))
+
+}
+
+
+dfc.strip_for_comparison <- function(DF){
+  # formats a DF to eliminate frequent causes of unimportant mismatches, including
+  # differently-formatted blanks, rounding errors, and special characters
+  clean_blanks <- function(DF){
+    DF %>%
+      util.apply_columns(function(x) ifelse(util.is_blank(x), NA, x)) %>%
+      util.apply_columns(function(x) if(all(is.na(x))) return(rep(NA, length(x))) else return(x))
```

Comment:
please-fix-or-comment: this line (156) seems like a noop ("non-operation" i.e. doesn't do anything).

Wouldn't any `x` that satisfies `all(is.na(x))` be identical to `rep(NA, length(x))`?